### ISSUE TITLE
Mill Wifi Socket

### ISSUE THREAD

**Issue by torke1** [glennsp/com.mill#26]
Feature wish: Can we get the Mill Wifi Socket working in the app? It doesn't seem to find it. Please?

**Comment by spajder77** [comment]
I have 5 WiFi sockets and they are working fine in homey.

**Comment by torke1** [comment]
That's weird. I have seen in other forums that people seem to have problem connecting to them. Could there be a bug? I have only one socket. No other Mill stuff.

**Comment by torke1** [comment]
I just tried reinstalling the app and re-entering the login. No luck.

**Comment by torke1** [comment]
Sorry for spamming. Just found the problem: The device needs to be assigned to a room. The Homey app cannot handle Independent Devices.

**Comment by martinju** [comment]
> Sorry for spamming. Just found the problem: The device needs to be assigned to a room. The Homey app cannot handle Independent Devices.

I also realized the same thing for my Mill oven (not socket). From what I understand this is, however, a serious problem. If one wishes to be able to use the oven like a "dumb" oven to be controlled by manually setting the thermostat directly on the physical oven, I believe the device cannot be assigned to a room, but actually needs to be an independent device. Agree? Is there a workaround, or a proper way to fix this issue?

**Comment by torke1** [comment]
Agree. I found a simple workaround though: First, add the socket to a room. Then add it to Homey. When this is done, you can remove it from the room again.

**Comment by martinju** [comment]
@torke1 Thanks for the rapid reply, but unfortunately this does not work for me. When I remove the heater from the room, it does not respond to changes in homey anymore. When I add it back it works as before. I got the Mill SG1200WIFI heater, not just the socket.

**Comment by torke1** [comment]
Weird. I think I didn't actually remove it from the room, but somehow opened the setting for the device and chose to make it independent. Still, I don't dare trying it again now that it works, so I'm afraid I cannot help you more than that.

I still think this is a bug, so I hope @glennsp could fix this.
